test/e2e/packetcapture_test.go

```diff
@@ -0,0 +1,728 @@
+					Containers: []v1.Container{
+						{
+							Name:            "sftp",
+							Image:           "antrea/sftp",
+							ImagePullPolicy: v1.PullIfNotPresent,
+							Args:            []string{"foo:pass:::upload"},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createUDPServerPod(name string, ns string, portNum int32, serverNode string) error {
+	port := v1.ContainerPort{Name: fmt.Sprintf("port-%d", portNum), ContainerPort: portNum}
+	return NewPodBuilder(name, ns, agnhostImage).
+		OnNode(serverNode).
+		WithContainerName("agnhost").
+		WithArgs([]string{"serve-hostname", "--udp", "--http=false", "--port", fmt.Sprint(portNum)}).
+		WithPorts([]v1.ContainerPort{port}).
+		Create(testData)
+}
+
+// TestPacketCapture is the top-level test which contains all subtests for
+// PacketCapture related test cases, so they can share setup, teardown.
+func TestPacketCapture(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	var previousAgentPacketCaptureEnableState bool
+	ac := func(config *agentconfig.AgentConfig) {
+		previousAgentPacketCaptureEnableState = config.FeatureGates[string(features.PacketCapture)]
+		config.FeatureGates[string(features.PacketCapture)] = true
+	}
+	if err := data.mutateAntreaConfigMap(nil, ac, false, true); err != nil {
+		t.Fatalf("Failed to enable PacketCapture flag: %v", err)
+	}
+	defer func() {
+		ac := func(config *agentconfig.AgentConfig) {
+			config.FeatureGates[string(features.PacketCapture)] = previousAgentPacketCaptureEnableState
+		}
+		if err := data.mutateAntreaConfigMap(nil, ac, false, true); err != nil {
+			t.Errorf("Failed to disable PacketCapture flag: %v", err)
+		}
+	}()
+
+	// setup sftp server for test.
+	secretUserName := "foo"
+	secretPassword := "pass"
+	_, err = data.clientset.AppsV1().Deployments(data.testNamespace).Create(context.TODO(), genSFTPDeployment(), metav1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = data.clientset.CoreV1().Services(data.testNamespace).Create(context.TODO(), genSFTPService(), metav1.CreateOptions{})
+	require.NoError(t, err)
+	failOnError(data.waitForDeploymentReady(t, data.testNamespace, "sftp", defaultTimeout), t)
+
+	sec := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pcSecretName,
+			Namespace: pcSecretNamespace,
+		},
+		Data: map[string][]byte{
+			"username": []byte(secretUserName),
+			"password": []byte(secretPassword),
+		},
+	}
+	_, err = data.clientset.CoreV1().Secrets(pcSecretNamespace).Create(context.TODO(), sec, metav1.CreateOptions{})
+	require.NoError(t, err)
+	defer data.clientset.CoreV1().Secrets(pcSecretNamespace).Delete(context.TODO(), pcSecretName, metav1.DeleteOptions{})
+
+	t.Run("testPacketCaptureBasic", func(t *testing.T) {
+		testPacketCaptureBasic(t, data)
+	})
+	t.Run("testPacketCapture", func(t *testing.T) {
+		testPacketCapture(t, data)
+	})
+
+}
+
+func testPacketCapture(t *testing.T, data *TestData) {
```

Comment:
I would suggest removing the more advanced case from this PR for now, as the PR is big enough
We can add it back as a separate PR after the release - cc @luolanzone @tnqn

Reply:
i merged them with the `basic` tests, these error-case tests is still useful and not so `advanced`,  i think, also it can helped with the concurrency tests ( 5 tests for now)